src/HexagonOptimize.cpp

```diff
@@ -269,6 +274,11 @@ Expr apply_patterns(Expr x, const vector<Pattern> &patterns, const Target &targe
                 continue;
             }
 
+            // Don't apply pattern if it does not involve an
```

Comment:
This comment is a bit confusing. I think it should be rewritten "Don't apply pattern if it involves an interleave, and it is not a multiple of two vectors" and link to the issue #1582.

Reply:
Done. Thanks.